### PR TITLE
Multiple restore sources

### PR DESCRIPTION
**Pull request by roend83**
I added the ability to pass an array of NuGet sources to the restore command.

### CODE REVIEW

File: test/js/restoreArgBuilder-test.js [hunk 1]
```diff
@@ -13,6 +13,9 @@ describe('Restore Argument Builder', () => {
     it('should have source if source is passed', () => {
       assert.deepEqual(builder({source: 'yo'}), ['--source', 'yo']);
     });
+    it('should have multiple sources if source array is passed', () => {
+      assert.deepEqual(builder({source: ['yo1', 'yo2']}), ['--source', 'yo1', '--source', 'yo2']);
```
Comment (TerribleDev) [inline test/js/restoreArgBuilder-test.js:17]: Thanks for following the yo naming convention

File: lib/builders/restoreArgBuilder.js [hunk 2]
```diff
@@ -4,7 +4,13 @@ module.exports = (value) => {
         return args;
     }
     if (value.source) {
-        args = args.concat(['--source', value.source]);
+        if (Array.isArray(value.source)) {
+            for (let source of value.source) {
```
Comment (TerribleDev) [inline lib/builders/restoreArgBuilder.js]: `args = args.concat(_.map(data, (a)=> ['--sources', a]))

File: lib/builders/restoreArgBuilder.js [hunk 3]
```diff
@@ -4,7 +4,13 @@ module.exports = (value) => {
         return args;
     }
     if (value.source) {
-        args = args.concat(['--source', value.source]);
+        if (Array.isArray(value.source)) {
```
Comment (TerribleDev) [inline lib/builders/restoreArgBuilder.js]: `_.isArray(value.source)` if you don't mind. Not that this is incorrect, but I'd prefer to use lodash for most array functions

File: lib/builders/restoreArgBuilder.js (hunk 2 above)
Comment (TerribleDev) [inline lib/builders/restoreArgBuilder.js]:
```js

args = _.isArray(value.source) ? args.concat(_.map(value.source, (a)=> ['--source', a])) : args.concat(['--source', value.source]);

```

**Comment by TerribleDev** [comment]
Thanks for the contribution @roend83 if you don't mind just using a ternary and some lodash (I gave pretty much the code above) I would appreciate it!

File: lib/builders/restoreArgBuilder.js (hunk 2 above)
Comment (TerribleDev) [reply lib/builders/restoreArgBuilder.js]: you may need to  add `var _ = require('lodash');`

**Comment by roend83** [comment]
_.map didn't work by itself.  I had to flatten out the result array.  Let me know if that looks better.

File: lib/builders/restoreArgBuilder.js [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
     if (value.source) {
-        if (Array.isArray(value.source)) {
-            for (let source of value.source) {
-                args = args.concat(['--source', source]);
-            }
-        } else {
-            args = args.concat(['--source', value.source]);
-        }
+        args = args.concat(_.isArray(value.source) 
+            ? _.chain(value.source).map((source) => ['--source', source]).flatten().value()
```
Comment (TerribleDev) [inline lib/builders/restoreArgBuilder.js]: oh I forgot about the flatten, boy that was a derp move on my part!

**Comment by TerribleDev** [comment]
@roend83 <3

**Comment by TerribleDev** [comment]
@roend83 0.4.0 https://www.npmjs.com/package/gulp-dotnet-cli

**Comment by TerribleDev** [comment]
its published

**Pull request merged**